xbmc/windowing/android/WinSystemAndroid.cpp

```diff
@@ -219,66 +217,48 @@ void CWinSystemAndroid::UpdateResolutions(bool bUpdateDesktopRes)
 
 void CWinSystemAndroid::OnTimeout()
 {
-  // We don't trigger OnResetDisplay if we wait for HDMI connect
-  {
-    CSingleLock lock(m_resourceSection);
-    if (m_dispResetState & RESET_WAIT_HDMIPLUG)
-    {
-      // Let HDMI_PLUG trigger the reset
-      m_dispResetState &= ~RESET_WAIT_HDMIPLUG;
-      return;
-    }
-  }
-  SetHDMIState(HDMI_STATE_CONNECTED);
+  SetHDMIState(true);
 }
 
-void CWinSystemAndroid::SetHDMIState(uint8_t state)
+void CWinSystemAndroid::InitiateModeChange()
 {
   CSingleLock lock(m_resourceSection);
-  CLog::Log(LOGDEBUG, "CWinSystemAndroid::SetHDMIState: state: %d, dispResetState: %d",
-            static_cast<int>(state), m_dispResetState);
   int delay = CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(
                   "videoscreen.delayrefreshchange") *
               100;
-  if (state & HDMI_STATE_CONNECTED)
+
+  if (delay < 2000)
+    delay = 2000;
+  m_dispResetTimer->Stop();
+  m_dispResetTimer->Start(delay);
+
+  SetHDMIState(false);
+}
+
+void CWinSystemAndroid::SetHDMIState(bool connected)
+{
+  CSingleLock lock(m_resourceSection);
+  CLog::Log(LOGDEBUG, "CWinSystemAndroid::SetHDMIState: state: %d", static_cast<int>(connected));
+
+  if (connected)
   {
-    if (m_dispResetState & RESET_WAIT_HDMIPLUG)
+    if (m_dispResetTimer->IsRunning())
     {
-      m_dispResetState &= ~RESET_WAIT_HDMIPLUG;
-
-      if (m_dispResetTimer->GetElapsedMilliseconds() >= delay)
-        // Most probably a pseudo (2 sec) timer -> signal now
+      if (m_dispResetTimer->GetElapsedMilliseconds() >=
```

Comment:
A comment here would be good that this is used for the "wakeup check", e.g. we exit here and stop the timer when it was just "misused" to wait for the initial hotplug enable.